core/src/main/resources/hudson/PluginManager/PluginDeprecationMonitor/message.jelly

```diff
@@ -0,0 +1,35 @@
+<!--
+The MIT License
+
+Copyright (c) 2019, Joseph Petersen
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:d="jelly:define" xmlns:l="/lib/layout" xmlns:t="/lib/hudson" xmlns:f="/lib/form">
+  <div class="alert alert-danger">
+    <dl>
+      <dt>${%DeprecatedPlugins}</dt>
+      <j:forEach var="p" items="${it.deprecatedPlugins}">
+        <dd><j:out value="${p.longName} v${p.version}"/></dd>
```

Comment:
Introduces an XSS vulnerability through plugin display names. Do not use `j:out` on untrusted input.

Reply:
What is the alternative? :sweat_smile:

Reply:
```suggestion
        <dd>${p.longName} v${p.version}</dd>
```

would be equivalent except for being not unsafe.

```suggestion
        <dd><a href="${p.url}" target="_blank">${p.longName} ${p.version}</dd>
```

would also link to the plugin docs (and remove the `v` before the version, something I don't think we're using in general?).

Reply:
p.url can be empty :cry:

Reply:
Then check for that and don't link if it's bad.

Also don't link if not http/https.

Reply:
What would you consider a bad link?